src/PostgREST/App.hs

```diff
@@ -241,11 +241,12 @@ app dbStructure conf apiRequest =
             Right ((q, cq), bField, params) -> do
               let prms = case payload of
                           Just (PayloadJSON pld) -> V.head pld
-                          Nothing -> M.fromList $ second toJSON <$> params -- toJSON is just for reusing the callProc function
+                          Nothing -> M.fromList $ second toJSON <$> params
                   singular = contentType == CTSingularJSON
                   paramsAsSingleObject = iPreferSingleObjectParameter apiRequest
-              row <- H.query () $
-                callProc qi prms returnsScalar q cq shouldCount
+                  specifiedPgArgs = filter (flip M.member prms . pgaName) $ fromMaybe [] (pdArgs <$> proc)
```

Comment:
So `specifiedPgArgs` strips away json keys that don't match the param names that postgrest has cached in dbStructure for the given proc?

Can postgresql overload a single function name with procedures taking different numbers of arguments, and if so would the logic here prevent selecting the desired proc through choice of arguments?

Reply:
My intention with that variable was to avoid specifying types for the parameters that aren't arguments so that functions with DEFAULT parameters work as normal, e.g. if the `filter` was not applied to `pdArgs` when calling this [function](https://github.com/begriffs/postgrest/blob/master/test/fixtures/schema.sql#L1292-L1294)  with `GET /rpc/many_inout_params?num=3&str=smth` it would give `[{"b": null,"num": 3,"str": "smth"}]` instead of the correct `[{"b": true, "num": 3, "str": "smth"}]`.

Now as you noticed this new way of specifying the arguments doesn't work for overloaded functions, I didn't realized this until now(tests would've made it clear, I'll add those if I get it to work), previous version was working due to the `::unknown` casting, for trying something similar in the new query I did:
```sql
with args as (
  select * from json_to_record('{"num":1, "str": "smth", "b": true}') as _(num unknown, str unknown, b unknown)
)
select test.many_inout_params(num := (select num from args), str := (select str from args), b:= (select b from args));
```
But postgres cannot infer the types and gives: `ERROR:  XX000: failed to find conversion function from unknown to integer`, so it cannot be done in that way.

For now, I'll be trying to specify the types as unknowns at the `hasql` level.

Reply:
Did a PR on hasql for handling this case, if it gets accepted it would not be necessary to use `json_to_record`, we would just use the `unknown` encoder for passing the args to the function.

Reply:
Cool!

Reply:
Woo hoo, looks like hasql 1.1 is now available and includes your change!

Reply:
while that is a good thing, in this case as we discussed bellow, to take advantage of that, we'd need to parse the json and maybe it makes sense to avoid that (then entire body as a single param).
what hasql 1.1 can help with (in the future) is to transition to real prepared statements.